pyfixest/estimation/decomposition.py

```diff
@@ -126,14 +125,22 @@ def fit(self, X: np.ndarray, Y: np.ndarray, store: bool = True):
                 self.beta2,
                 self.beta2_sparse,
                 self.contribution_dict,
-                self.X_dict,
-                self.Y_dict,
             ) = results
 
+            # Prepare cluster bootstrap if relevant
+            self.X_dict = {}
+            self.Y_dict = {}
+
+            if self.unique_clusters is not None:
+                for g in self.unique_clusters:
+                    cluster_idx = np.where(self.cluster_df == g)[0]
+                    self.X_dict[g] = self.X[cluster_idx]
+                    self.Y_dict[g] = self.Y[cluster_idx]
+
             return self.contribution_dict
 
         else:
-            X1 = np.concatenate([np.ones((self.N, 1)), X[:, ~self.mask]], axis=1)
+            X1 = np.concatenate([np.ones((X.shape[0], 1)), X[:, ~self.mask]], axis=1)
```

Comment:
was this the source of the error? i don't understand why the bootstrap loop would require a different data matrix from the point estimation flow

Reply:
No, that wasn't it - I just pushed the bug fix with [accf066](https://github.com/py-econometrics/pyfixest/pull/669/commits/accf066c6999606c47d41c18c2320f1aaf07c328). Basically dropped an intercept when I shouldn't have.

Reply:
To save you some time the issue was to do 

```python
            delta = np.linalg.lstsq(X1[:, 1:], Hg, rcond=None)[0].flatten()
```
instead of 
```python
            delta = np.linalg.lstsq(X1, Hg, rcond=None)[0].flatten()
```

Reply:
> i don't understand why the bootstrap loop would require a different data matrix from the point estimation flow
I actually think this is needed, but will take another look tomorrow. Too late in the evening over here to think this through ...

Reply:
Actually you're right and adjusted with [43b8e79](https://github.com/py-econometrics/pyfixest/pull/669/commits/43b8e79ae46717b1b2b674688fc9a023ca63c93b). Thanks!

Reply:
I actually had to revert it - the reason is that bootstrapped X and Y are passed to fit and X1 and X2 need to be created from the bootstrapped X1.